parlai/core/agents.py

```diff
@@ -131,6 +131,40 @@ def shutdown(self):
         """Perform any final cleanup if needed."""
         pass
 
+    @classmethod
+    def upgrade_opt(cls, opt_from_disk):
+        """
+        Upgrade legacy options when loading an opt file from disk.
+
+        When loading a previously trained model file, we may wish to ensure
+        backwards compatibility. For example, we may introduce a new option
+        today that wasn't available initially, but we want the default on all
+        new models to be the new behavior.
```

Comment:
Super nitty doc comment with extended reasoning: This feels a little unclear as it's talking about what the intended effect of `upgrade_opts` implies (all new models can happily use the new default) without explicitly stating the requirement that `upgrade_opts` should satisfy for older models ("To keep the behavior of old models consistent, `upgrade_opts` should disable the new option if it's not present in the loaded opts."). I think this is partially implied by "backwards compatibility," but that term can also accurately be read as "still works without breaking" rather than "works in exactly the way it used to."